src/parallel/parallel.hpp

```diff
@@ -131,9 +132,13 @@ class SegmentLock {
 /// HELPER CLASSES
 ////////////////////////////////////////////////////////////////////////////////
 
-struct ParallelBucket {
+struct KeyValue {
```

Comment:
Are you sure the packing will be as expected? i.e. there won't be any padding so that sizeof(KeyValue) == 8 or 16 bytes (depending on what we decide to use for our Key/Value types).